codecov_cli/parsers/base.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime, timedelta
+from typing import List
+
+
+class ParsingError(Exception):
+    ...
+
+
+class Test(object):
+    def __init__(self, name: str, status: bool, duration: timedelta):
+        self.name = name
+        self.status = status
+        self.duration = duration
+
+    def __repr__(self):
+        return f"{self.name}:{self.status}:{self.duration.total_seconds()}::"
```

Comment:
Is this to be used as the encoding/decoding or just debugging?
If just for debugging than it's very confusing.

If it's for the encoding/decoding I think it's ok to use JSON directly.
I understand that the payload will be larger, but we compress it before uploading. And making the data easier to understand and work with seems more relevant... I could be wrong, if you have data backing up the fact that unless we encode the data in this compact way the resulting payload will be huge.